android-lib/src/main/java/com/kinvey/android/store/UserStore.java

```diff
@@ -1116,7 +1116,9 @@ protected T executeAsync() throws IOException {
             UserStoreRequestManager requestManager = new UserStoreRequestManager(client, createBuilder(client));
             requestManager.setMICRedirectURI(redirectURI);
             GenericJson result = requestManager.getMICToken(token, clientId).execute();
-
+            if (result.get(ACCESS_TOKEN) == null || result.get(REFRESH_TOKEN) == null) {
```

Comment:
It is allowed that you can have an `access_token` but not necessarily a `refresh_token`, so we cannot have this `||` logic here.

Reply:
Fixed